ra-optimisation/rao-api/src/main/java/com/farao_community/farao/rao_api/RaoTool.java

```diff
@@ -112,20 +109,23 @@ public void run(CommandLine line, ToolRunningContext context) throws Exception {
         Network network = Importers.loadNetwork(caseFile);
         String currentState = network.getVariantManager().getWorkingVariantId();
 
-        //Crac for SearchTreeRao
+        //Crac
+        context.getOutputStream().println("Importing crac '" + cracFile + "'");
         Crac crac = CracImporters.importCrac(cracFile);
 
         //Rao Parameter
+        context.getOutputStream().println("Loading RAO parameters");
         RaoParameters raoParameters = RaoParameters.load(PlatformConfig.defaultConfig());
 
         //Run
         ComputationManager computationManager = context.getLongTimeExecutionComputationManager();
-        CompletableFuture<RaoComputationResult> raoComputationResult = new SearchTreeRao().run(network, crac, currentState, computationManager, raoParameters);
-        SearchTreeRaoResult searchTreeRaoResult = raoComputationResult.get().getExtension(SearchTreeRaoResult.class);
+        context.getOutputStream().println("Running Rao computation");
+        RaoComputationResult raoComputationResult = Rao.run(network, crac, currentState, computationManager, raoParameters);
+        RaoComputationResult raoResult = raoComputationResult.getExtension(RaoComputationResult.class);
```

Comment:
We should return directly the raoComputationResult. If you take only `raoComputationResult.getExtension(RaoComputationResult.class)` I'm afraid we might loose extension related to the different Rao implementations

Reply:
corrected